aea/cli_gui/__init__.py

```diff
@@ -85,6 +85,24 @@ class AppContext:
 app_context = AppContext()
 
 
+def _call_subprocess(*args, **kwargs):
+    """
+    Call subprocess.call, but with error handling.
+
+    :return the exit code, or -1 if the call raises exception.
+    """
+    try:
+        ret = subprocess.call(*args, **kwargs)  # nosec
+    except BaseException:
+        logging.exception(
+            "An exception occurred when calling with args={} and kwargs={}".format(
+                args, kwargs
+            )
+        )
+        return -1
```

Comment:
My concern is that the places which call `call_aea/call_aea_async` don't necessarily close the subprocess. Shouldn't there be a try finally clause wrapping each of them:
```suggestion
    try:
        ret = call_aea(..)
    finally:
        poll = ret.poll()
        if poll is None:  # pragma: no cover
            ret.terminate()
            ret.wait(2)
```